types/next/router.d.ts

```diff
@@ -32,5 +32,7 @@ export interface SingletonRouter {
     onRouteChangeError?(error: any, url: string): void;
 }
 
+export function withRouter<T extends {}>(Component: React.ComponentType<T & {router: SingletonRouter}>): React.ComponentType<T>;
```

Comment:
I'm unsure about this `T extends {}` syntax.  Any reason not to just use `T`?  Does this guarantee that it can be any shape of object?

Reply:
Yeah. It's a pattern I've seen used in many React HOCs when annotated with TypeScript. But I don't have a strong opinion on it. I'm okay if it needs to go.

Reply:
It's technically redundant in Typescript, but not harmful. Maybe it's meaningful in flow?